ballerina/tests/constraint_annotation_tests.bal

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License
+
+import ballerina/constraint;
+import ballerina/test;
+import ballerina/time;
+
+public type ValidationPerson record {|
+    @constraint:String {
+        maxLength: 5,
+        minLength: 2
+    }
+    string name;
+    @constraint:Int {
+        minValueExclusive: 5
+    }
+    int age;
+    @constraint:Float {
+        minValue: 150,
+        maxValue: 185.20,
+        maxFractionDigits: 2
+    }
+    float height;
+    @constraint:Date {
+        option: {
+            value: "PAST",
+            message: "Date of birth should be past value"
+        },
+        message: "Invalid date found for date of birth"
+    }
+    time:Date dob;
+|};
+
+@test:Config {
+    groups: ["constraint-validation"]
+}
+function testValidConstraintAnnotationForParseString() returns error? {
+    string jsonStr = string `{
+        "name": "John",
+        "age": 6,
+        "height": 180.20,
+        "dob": {
+            "year": 1990,
+            "month": 12,
+            "day": 31
+        }
+    }`;
+
+    ValidationPerson person = check parseString(jsonStr);
+    test:assertEquals(person.name, "John");
+    test:assertEquals(person.age, 6);
+    test:assertEquals(person.height, 180.20);
+    test:assertEquals(person.dob.year, 1990);
+    test:assertEquals(person.dob.month, 12);
+    test:assertEquals(person.dob.day, 31);
+}
+
+@constraint:Array {
+    length: 2
+}
+public type Weight decimal[];
+
+public type ValidationItem record {|
+    Weight weight;
+|};
+
+@test:Config {
+    groups: ["constraint-validation"],
+    dataProvider: invalidConstraintAnnotation
+}
+function testInvalidConstraintAnnotationForParseString(string sourceData, typedesc<record {}> expType, string expectedError) {
```

Comment:
Add a test for disabling constraint validation for same samples

Reply:
Added in [commit](https://github.com/ballerina-platform/module-ballerina-data.yaml/pull/4/commits/8b27ffb0f928ac5be6f8bd55a3cda9217eff8ac4).